UpdateCgnOrchestrator/handler.ts

```diff
@@ -117,6 +149,36 @@ export const UpdateCgnOrchestratorHandler = function*(
           "cgn.update.exception.failure.activityOutput"
         );
       }
+      if (newStatusCard.status === ActivatedStatusEnum.ACTIVATED) {
+        const upsertSpecialServiceActiveActivityInput = UpsertSpecialServiceActivationActivityInput.encode(
+          {
+            activationStatus: ActivationStatusEnum.ACTIVE,
+            fiscalCode
+          }
+        );
+
+        const upsertSpecialServiceResult = pipe(
+          yield context.df.callActivityWithRetry(
+            "UpsertSpecialServiceActivationActivity",
+            internalRetryOptions,
+            upsertSpecialServiceActiveActivityInput
+          ),
+          ActivityResult.decode,
+          E.getOrElseW(e =>
+            trackExAndThrowWithError(
+              e,
+              "cgn.update.exception.upsertSpecialServiceActive.activityOutput"
+            )
+          )
+        );
+
+        if (upsertSpecialServiceResult.kind !== "SUCCESS") {
+          trackExAndThrowWithError(
+            new Error("Cannot upsert CGN Special service activation"),
+            "cgn.update.exception.failure.upsertSpecialService.activityOutput"
+          );
+        }
```

Comment:
What do you think about this more functional version?
```suggestion
        pipe(
          yield context.df.callActivityWithRetry(
            "UpsertSpecialServiceActivationActivity",
            internalRetryOptions,
            upsertSpecialServiceActiveActivityInput
          ),
          ActivityResult.decode,
          E.getOrElseW(e =>
            trackExAndThrowWithError(
              e,
              "cgn.update.exception.upsertSpecialServiceActive.activityOutput"
            )
          ),
          E.fromPredicate(
            _ => _.kind === "SUCCESS",
            () =>
              trackExAndThrowWithError(
                new Error("Cannot upsert CGN Special service activation"),
                "cgn.update.exception.failure.upsertSpecialService.activityOutput"
              )
          )
        );
```